app/controllers/requests.js

```diff
@@ -0,0 +1,33 @@
+(function(exports) {
+  'use strict';
+
+  exports.RequestController = function(promoListModel, promoListView, RequestClass) {
+    var _promoListModel = promoListModel;
+    var _promoListView = promoListView;
+    var _listController = new ListController(_promoListView);
+
+    function doRequest() {
+      var _httpRequest = new RequestClass();
+      _httpRequest.onreadystatechange = function() {
+        if (_httpRequest.readyState === RequestClass.DONE) {
+          if (_httpRequest.status === 200) {
+            var results = JSON.parse(_httpRequest.responseText).response.results;
+            updatePromoListModel(results);
+          }
+        }
+      };
+      _httpRequest.open("GET", "http://news-summary-api.herokuapp.com/guardian?apiRequestUrl=http://content.guardianapis.com/politics?show-fields=all", true);
```

Comment:
Try to avoid hard-coding the URL - it means the code is harder to re-use. I realise there's only one API in this app, but good practice would to be to have a generate 'doRequest' method that accepted the verb (GET, POST, etc), the url, and any post data, and then have a separate class responsible for defining the API.